### ISSUE TITLE
Unable to configure vite as suggested by official docs for monorepos

### ISSUE THREAD

**Issue by BernhardSchlegel** [quasarframework/quasar#15485]
### What happened?

I have a monorepo with a quasar-app and a shared library. Shared library is listed under dependencies in package.json:

```
  "dependencies": {
    "fooshared": "../shared/dist",
  }
```
I suppose I import a class from this lib as follows

```
import { SayHello } from "fooshared/user";
```
I'm getting an error when I do `quasar build` 

```
Error: [vite]: Rollup failed to resolve import "fooshared/user" from "src/layouts/MainLayout.vue".
```
When I do `quasar serve` the Web UI displays

```
[plugin:vite:import-analysis] Failed to resolve import "fooshared/user" from "src/layouts/MainLayout.vue". Does the file exist?
```

To mitigate that, the [vite documentation](https://vitejs.dev/guide/dep-pre-bundling.html#monorepos-and-linked-dependencies) suggests

```
export default defineConfig({
  optimizeDeps: {
    include: ['linked-dep'],
  },
  build: {
    commonjsOptions: {
      include: [/linked-dep/, /node_modules/],
    },
  },
})
```
so following the [quasar documentation regarding vite config](https://quasar.dev/quasar-cli-vite/handling-vite#usage-with-quasar-config-js) I added the following lines to my quasar config

```
// quasar.conf.js
viteConf.optimizeDeps.include = ["fooshared"];
viteConf.build = {
  commonjsOptions: {
    include: [/fooshared/, /node_modules/],
  },
}
```

(doing `viteConf.build.commonjsOptions.include = [/bbshared/, /node_modules/]` was not possible because it gave the error `Cannot set properties of undefined (setting 'include')
`)

But this gives **THE FINAL REASON FOR THIS BUG** error - see log below.

### What did you expect to happen?

No error build completing as expected.

### Reproduction URL

https://github.com/BernhardSchlegel/quasar-debug

### How to reproduce?

1. Create a new Quasar project
2. add a local / linked dependecy 
3. `quasar build`

### Flavour

Quasar CLI with Vite (@quasar/cli | @quasar/app-vite)

### Areas

Quasar CLI Commands/Configuration (@quasar/cli | @quasar/app-webpack | @quasar/app-vite)

### Platforms/Browsers

Firefox, Chrome, Safari, Microsoft Edge, iOS, Android, Electron, Other

### Quasar info output

```Shell
Operating System - Linux(5.19.0-32-generic) - linux/x64
NodeJs - 18.14.2

Global packages
  NPM - 9.5.0
  yarn - 1.22.19
  @quasar/cli - undefined
  @quasar/icongenie - Not installed
  cordova - Not installed

Important local packages
  quasar - 2.11.6 -- Build high-performance VueJS user interfaces (SPA, PWA, SSR, Mobile and Desktop) in record time
  @quasar/app-vite - 1.2.0 -- Quasar Framework App CLI with Vite
  @quasar/extras - 1.15.11 -- Quasar Framework fonts, icons and animations
  eslint-plugin-quasar - Not installed
  vue - 3.2.47 -- The progressive JavaScript framework for building modern web UI.
  vue-router - 4.1.6
  pinia - 2.0.32 -- Intuitive, type safe and flexible Store for Vue
  vuex - Not installed
  vite - 2.9.15 -- Native-ESM powered web dev build tool
  eslint - 8.34.0 -- An AST-based pattern checker for JavaScript.
  electron - Not installed
  electron-packager - Not installed
  electron-builder - Not installed
  register-service-worker - 1.7.2 -- Script for registering service worker, with hooks
  @capacitor/core - Not installed
  @capacitor/cli - Not installed
  @capacitor/android - Not installed
  @capacitor/ios - Not installed

Quasar App Extensions
  *None installed*

Networking
  Host - Ubuntu-22
  enp1s0 - 192.168.178.70
```

### Relevant log output

```Shell
/home/bernhard/quasar-project-3/node_modules/@quasar/app-vite/lib/artifacts.js:44
  if (folder.endsWith(path.join('src-cordova', 'www'))) {
             ^

TypeError: Cannot read properties of undefined (reading 'endsWith')
    at module.exports.clean (/home/bernhard/quasar-project-3/node_modules/@quasar/app-vite/lib/artifacts.js:44:14)
    at SpaBuilder.buildWithVite (/home/bernhard/quasar-project-3/node_modules/@quasar/app-vite/lib/app-tool.js:18:15)
    at SpaBuilder.build (/home/bernhard/quasar-project-3/node_modules/@quasar/app-vite/lib/modes/spa/spa-builder.js:8:16)
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
```

### Additional context

Maybe it's already obvious. If not I can fill the above repo with a monorepo setup.

**Comment by BernhardSchlegel** [comment]
Hey there, just letting y'all know: The repo linked above is filled with a minimal example of a quasar project and a lokal linked dependency to surface the error.

So I think we can drop the "bug/1-hard-to-reproduce" tag as it boils down to "git clone" and "quasar build".

Thanks in advance!

**Comment by freddy38510** [comment]
I think the error is due to the fact that you have overwritten the `build` property.

```javascript
//  set the `commonjsOptions` property first
if (!viteConf.build.commonjsOptions) { viteConf.build.commonjsOptions = {}; }

// then set the `include` property
viteConf.build.commonjsOptions.include =  [/fooshared/, /node_modules/];

```

**Comment by BernhardSchlegel** [comment]
Thanks so much for having a look! So I have this now 

```      extendViteConf (viteConf, { isServer, isClient }) {
        viteConf.optimizeDeps.include = ["fooshared"];

        //  set the `commonjsOptions` property first
        if (!viteConf.build.commonjsOptions) { viteConf.build.commonjsOptions = {}; }

        // then set the `include` property
        viteConf.build.commonjsOptions.include =  [/fooshared/, /node_modules/];
      },
```

But still no luck. Before creating this issue I printed `viteConf.build.commonjsOptions` to console and it was always `undefined`.

**Comment by freddy38510** [comment]
@BernhardSchlegel I guess the same logic should be applied to `viteConf.optimizeDeps.include`:

```javascript
viteConf.optimizeDeps = viteConf.optimizeDeps || {};
viteConf.optimizeDeps.include = viteConf.optimizeDeps.incude || [];
viteConf.optimizeDeps.include.push('fooshared');

viteConf.build.commonjsOptions = viteConf.build.commonjsOptions || {};
viteConf.build.commonjsOptions.include = viteConf.build.commonjsOptions.include || [];
viteConf.build.commonjsOptions.include.push(/fooshared/, /node_modules/);
```

**Comment by BernhardSchlegel** [comment]
Unfortunately, no luck either:

```
[vite]: Rollup failed to resolve import "fooshared/user" from "src/pages/IndexPage.vue".
This is most likely unintended because it can break your application at runtime.
If you do want to externalize this module explicitly add it to
`build.rollupOptions.external`
/home/bernhard/Desktop/wd/deleteme/quasar-debug/quasar-project/node_modules/vite/dist/node/chunks/dep-689425f3.js:41797
            throw new Error(`[vite]: Rollup failed to resolve import "${id}" from "${importer}".\n` +
                  ^

Error: [vite]: Rollup failed to resolve import "fooshared/user" from "src/pages/IndexPage.vue".
This is most likely unintended because it can break your application at runtime.
If you do want to externalize this module explicitly add it to
`build.rollupOptions.external`
```

when performing `quasar build`.

**Comment by freddy38510** [comment]
Dependency pre-bundling only applies in development mode.

The issue here is the resolution of the package `fooshared`. Looking at your reproduction url, it looks like you have misconfigured the monorepo. I'm going to create a PR to try to help you.

**Comment by freddy38510** [comment]
[PR](https://github.com/BernhardSchlegel/quasar-debug/pull/1) is available here.

**Comment by BernhardSchlegel** [comment]
Thank you soooooo much! Already checked out your comments. Your a quasar contributor? I tried to sponsor you :)

**Comment by freddy38510** [comment]
I really appreciate your intention to sponsor me.

I've made only [2 contributions](https://github.com/quasarframework/quasar/pulls?q=is%3Apr+author%3Afreddy38510+is%3Aclosed) to quasar. I also created, and still maintain, an [ssg extension](https://github.com/freddy38510/quasar-app-extension-ssg) for quasar.

I don't know how sponsorship works, so I looked at the github documentation. I guess I have to join the waitlist to be able to receive sponsors, which I did, I am now waiting for github approval.
